### PR TITLE
BAN-105: Dropdown: Use zen-menu-item in zen-dropdown

### PR DESCRIPTION
**Pull request by chavlji**
See [jira](https://reciprocitylabs.atlassian.net/browse/BAN-105) for detailed description.

Screenshot 2020-12-11 at 09 20 12

### CODE REVIEW

File: src/components/zen-dropdown/zen-dropdown.tsx [hunk 1]
```diff
@@ -31,19 +36,59 @@ export class ZenDropdown {
...
+    if (opened) {
+      if (!this.clickHandler) {
+        this.clickHandler = event => this.closeOnClickOut(event);
+      }
+      document.addEventListener('mousedown', this.clickHandler);
+
+      // Reset scroll:
+      if (this.list) {
+        await waitNextFrame();
+        await waitNextFrame();
```
Comment (f-jauregui) [inline src/components/zen-dropdown/zen-dropdown.tsx:80]: Are these two necessary?

File: src/components/zen-dropdown/zen-dropdown.tsx [hunk 2]
```diff
@@ -1,9 +1,10 @@
-import { Component, Host, h, Prop, State, Event, EventEmitter, Listen, Watch } from '@stencil/core';
+import { Component, Host, h, Prop, State, Event, EventEmitter, Listen, Watch, Element, Method } from '@stencil/core';
 import { key } from '../helpers/keyCodes';
 import { MouseEvent } from '../helpers/helpers';
 import { faChevronDown } from '@fortawesome/pro-light-svg-icons';
 import { renderIcon, styles } from '../helpers/fa-icons';
 import { get } from 'lodash';
```
Comment (f-jauregui) [inline src/components/zen-dropdown/zen-dropdown.tsx:6]: Please take care of lodash, it bloats the component at build time. Also I don't think you use get in this component.

File: src/components/zen-dropdown/zen-dropdown.tsx [hunk 3]
```diff
@@ -31,19 +36,59 @@ export class ZenDropdown {
...
   /** Emitted on any selection change */
   @Event() zenInput: EventEmitter<OptionValue>;
+  /** Focused item changed (keyboard arrows) */
+  @Event() zenFocusItem: EventEmitter<OptionValue>;
+
+  /** Close an opened dropdown menu */
+  @Method()
+  async close(): Promise<void> {
+    this.opened = false;
+  }
```
Comment (f-jauregui) [inline src/components/zen-dropdown/zen-dropdown.tsx:51]: If I open de dropdown by clicking the chevron and I want to close it clicking the same chevron it's not working. 
Can just be closed by selecting an item or clicking outside the dropdown.
Maybe a toggle function would be more useful than a close function.

File: src/components/zen-menu-item/zen-menu-item.tsx [hunk 4]
```diff
@@ -9,14 +9,16 @@ export class ZenMenuItem {
   /** Text inside the item */
   @Prop() readonly label: string = 'Item';
   /** Render item as selected */
-  @Prop() readonly selected: boolean = false;
+  @Prop({ reflect: true }) readonly selected: boolean = false;
+  /** Render item as focused */
+  @Prop({ reflect: true }) readonly focused: boolean = false;
   /** False to enable custom item padding */
   @Prop() readonly defaultPadding: boolean = true;
```
Comment (f-jauregui) [inline src/components/zen-menu-item/zen-menu-item.tsx:16]: These props will became useful with the next PR with the slots, right?

File: src/components/zen-dropdown/zen-dropdown.tsx [hunk 5]
```diff
@@ -1,9 +1,10 @@
-import { Component, Host, h, Prop, State, Event, EventEmitter, Listen, Watch } from '@stencil/core';
+import { Component, Host, h, Prop, State, Event, EventEmitter, Listen, Watch, Element, Method } from '@stencil/core';
 import { key } from '../helpers/keyCodes';
 import { MouseEvent } from '../helpers/helpers';
 import { faChevronDown } from '@fortawesome/pro-light-svg-icons';
 import { renderIcon, styles } from '../helpers/fa-icons';
 import { get } from 'lodash';
+import { waitNextFrame } from '../helpers/helpers';
```
Comment (f-jauregui) [inline src/components/zen-dropdown/zen-dropdown.tsx:7]: if waitNextFrame isn't necessary you won't need this import either.

File: src/components/zen-dropdown/zen-dropdown.tsx [hunk 6]
```diff
@@ -31,19 +36,59 @@ export class ZenDropdown {
...
+        this.markSelectedSlottedOption(this.value);
+      }
+    } else {
+      document.removeEventListener('mousedown', this.clickHandler);
+    }
+  }
+
   @Listen('keydown')
   handleKeyDown(ev: KeyboardEvent): void {
     const toggleKeys = [key.SPACE, key.ENTER, key.UP, key.DOWN];
```
Comment (f-jauregui) [inline src/components/zen-dropdown/zen-dropdown.tsx:94]: Take a look at this. Above is ev.keyCode which is deprecated: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode

**Review by f-jauregui — CHANGES_REQUESTED** [review]
Just those little changes.
Otherwise it's looking really good! 👍

File: src/components/zen-dropdown/zen-dropdown.tsx (hunk 2 above)
Comment (chavlji) [reply src/components/zen-dropdown/zen-dropdown.tsx:6]: Added ticket:
https://reciprocitylabs.atlassian.net/browse/BAN-109

File: src/components/zen-dropdown/zen-dropdown.tsx (hunk 1 above)
Comment (chavlji) [reply src/components/zen-dropdown/zen-dropdown.tsx:80]: Jup. If you remove them, it won't scroll back to the top when you open the list.
We have to wait for the list to actually appear. It takes 2 render cycles.

File: src/components/zen-menu-item/zen-menu-item.tsx (hunk 4 above)
Comment (chavlji) [reply src/components/zen-menu-item/zen-menu-item.tsx:16]: They're already useful and used for default zen-menu-item in zen-dropdown.

File: src/components/zen-dropdown/zen-dropdown.tsx (hunk 5 above)
Comment (chavlji) [reply src/components/zen-dropdown/zen-dropdown.tsx:7]: It has to stay 😢

**Review by f-jauregui — APPROVED** [review]

**Pull request merged**